tests/pytorch/test_fused_router.py

```diff
@@ -148,11 +148,15 @@ def run_comparison(
 ):
     # Set some parameters
     if score_function == "sigmoid":
-        logits = torch.arange(num_experts, device="cuda", dtype=dtype) * 0.1
+        # Construct the special logits to avoid inf in the sigmoid function
+        logits = 1 + torch.arange(num_experts, device="cuda", dtype=dtype) * 0.01
         logits = logits.unsqueeze(0).repeat(num_tokens, 1)
+        random_values = torch.rand(num_tokens, num_experts, device="cuda")
+        _, indices = torch.sort(random_values, dim=1)
+        logits = torch.gather(logits, 1, indices)
```

Comment:
Can you explain how this helps avoid infs? Was the problem that `logits` became too large, causing sigmoid to output 1.0 in multiple entries? Also, I don't understand the benefit from forcing `logits >= 1`. Previously we had `logits >= 0` and `sigmoid(0) = 0.5`, so I wouldn't expect us to have had divide-by-zero problems.